core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala

```diff
@@ -105,7 +105,11 @@ class NettyBlockRpcServer(
         val blockId = BlockId(uploadBlock.blockId)
         logDebug(s"Receiving replicated block $blockId with level ${level} " +
           s"from ${client.getSocketAddress}")
-        blockManager.putBlockData(blockId, data, level, classTag)
+        val blockStored = blockManager.putBlockData(blockId, data, level, classTag)
+        if (!blockStored) {
+          throw new Exception(s"Upload block for $blockId failed. This " +
```

Comment:
will it crash the spark application?

Reply:
No. Raising this exception on remote block manager helps in invoking onFailure callback on client block manager which inturn propogates to [replicate method](https://github.com/apache/spark/blob/v3.0.0-preview2-rc2/core/src/main/scala/org/apache/spark/storage/BlockManager.scala#L1626). And it retries to other peers for block replication after that.

Raising exception here is just equivalent to  "responseContext.ononFailure(e)".